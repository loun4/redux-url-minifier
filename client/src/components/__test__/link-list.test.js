import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LinkList from '../link-list';

const links = [
  {
    id: 'anId',
    linkURL: 'http://www.google.fr',
    shortURL: '/link/anId',
    meta: {
      created: null,
    },
  },
  {
    id: 'anId2',
    linkURL: 'http://www.github.com',
    shortURL: '/link/anId2',
    meta: {
      created: null,
    },
  },
];

let subject;

beforeEach(() => {
  const props = {
    link: {
      isFetching: false,
      isSaving: false,
      models: links,
    },
    onRemove: jest.fn(),
  };

  subject = <LinkList {...props} />;
});

test('Render list of links', () => {
  const component = renderer.create(subject);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Update table with received new links', () => {
  const component = mount(subject);
  expect(component.find('.rt-tr-group').length).toEqual(2);

  const link = {
    isFetching: false,
    isSaving: false,
    models: links.concat(links),
  };

  component.setProps({ link });
  expect(component.find('.rt-tr-group').length).toEqual(4);
});
