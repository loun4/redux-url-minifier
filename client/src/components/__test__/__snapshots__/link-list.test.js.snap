// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render list of links 1`] = `
<div
  className="ReactTable"
  style={Object {}}
>
  <div
    className="rt-table"
    role="grid"
    style={undefined}
  >
    <div
      className="rt-thead -header"
      style={
        Object {
          "minWidth": "270px",
        }
      }
    >
      <div
        className="rt-tr"
        role="row"
        style={undefined}
      >
        <div
          className="rt-th  rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Short URL
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th  rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "100 0 auto",
              "maxWidth": null,
              "width": "100px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Long URL
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th  rt-resizable-header -sort-desc -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "25 0 auto",
              "maxWidth": null,
              "width": "25px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Created at
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th  rt-resizable-header -cursor-pointer"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "15 0 auto",
              "maxWidth": null,
              "width": "15px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          >
            Visits
          </div>
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
        <div
          className="rt-th  rt-resizable-header"
          onClick={[Function]}
          role="columnheader"
          style={
            Object {
              "flex": "30 0 auto",
              "maxWidth": null,
              "width": "30px",
            }
          }
          tabIndex="-1"
        >
          <div
            className="rt-resizable-header-content"
          />
          <div
            className="rt-resizer"
            onMouseDown={[Function]}
            onTouchStart={[Function]}
          />
        </div>
      </div>
    </div>
    <div
      className="rt-tbody"
      style={
        Object {
          "minWidth": "270px",
        }
      }
    >
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -odd"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <div
              className="ui buttons mini"
            >
              <a
                className="ui button"
                href="/link/anId2"
                target="_blank"
              >
                <i
                  className="external alternate icon"
                />
                /link/anId2
              </a>
              <div
                className="or"
              />
              <button
                className="ui vertical green button"
                onClick={[Function]}
              >
                Copy
              </button>
            </div>
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            http://www.github.com
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "25 0 auto",
                "maxWidth": null,
                "width": "25px",
              }
            }
          >
            Invalid Date
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "15 0 auto",
                "maxWidth": null,
                "width": "15px",
              }
            }
          />
          <div
            className="rt-td rt-delete"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "30 0 auto",
                "maxWidth": null,
                "width": "30px",
              }
            }
          >
            <button
              className="negative ui button mini"
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
      <div
        className="rt-tr-group"
        role="rowgroup"
      >
        <div
          className="rt-tr -even"
          role="row"
          style={undefined}
        >
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            <div
              className="ui buttons mini"
            >
              <a
                className="ui button"
                href="/link/anId"
                target="_blank"
              >
                <i
                  className="external alternate icon"
                />
                /link/anId
              </a>
              <div
                className="or"
              />
              <button
                className="ui vertical green button"
                onClick={[Function]}
              >
                Copy
              </button>
            </div>
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "100 0 auto",
                "maxWidth": null,
                "width": "100px",
              }
            }
          >
            http://www.google.fr
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "25 0 auto",
                "maxWidth": null,
                "width": "25px",
              }
            }
          >
            Invalid Date
          </div>
          <div
            className="rt-td"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "15 0 auto",
                "maxWidth": null,
                "width": "15px",
              }
            }
          />
          <div
            className="rt-td rt-delete"
            onClick={[Function]}
            role="gridcell"
            style={
              Object {
                "flex": "30 0 auto",
                "maxWidth": null,
                "width": "30px",
              }
            }
          >
            <button
              className="negative ui button mini"
              onClick={[Function]}
            >
              Delete
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pagination-bottom"
  >
    <div
      className="-pagination"
      style={undefined}
    >
      <div
        className="-previous"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Previous
        </button>
      </div>
      <div
        className="-center"
      >
        <span
          className="-pageInfo"
        >
          Page
           
          <div
            className="-pageJump"
          >
            <input
              onBlur={[Function]}
              onChange={[Function]}
              onKeyPress={[Function]}
              type="number"
              value={1}
            />
          </div>
           
          of
           
          <span
            className="-totalPages"
          >
            1
          </span>
        </span>
        <span
          className="select-wrap -pageSizeOptions"
        >
          <select
            onChange={[Function]}
            value={20}
          >
            <option
              value={5}
            >
              5
               
              rows
            </option>
            <option
              value={10}
            >
              10
               
              rows
            </option>
            <option
              value={20}
            >
              20
               
              rows
            </option>
            <option
              value={25}
            >
              25
               
              rows
            </option>
            <option
              value={50}
            >
              50
               
              rows
            </option>
            <option
              value={100}
            >
              100
               
              rows
            </option>
          </select>
        </span>
      </div>
      <div
        className="-next"
      >
        <button
          className="-btn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
  <div
    className="-loading"
  >
    <div
      className="-loading-inner"
    >
      Loading...
    </div>
  </div>
</div>
`;
